UOP1_Project/Assets/Scripts/Characters/Character.cs

```diff
@@ -27,6 +32,7 @@ public class Character : MonoBehaviour
     private void Awake()
     {
         characterController = GetComponent<CharacterController>();
```

Comment:
Preferably just serialized the CharacterController, GetComponent is more costly!

Reply:
You are right, but I did not write that, it was already there...

Reply:
I disagree @TJHeuvel . Using GetComponent<T> in Awake or Start is largely accepted best practice when caching references for use later. The performance difference between setting the variable in the inspector, and using GetComponent<T> is negligible. It's only when you're using GetComponent<T> frequently, for example in the Update loop that it becomes a performance issue. 

For the character script, Awake should only be getting called maybe a couple of times over the life cycle of the scene; once on scene load, and any time the character gameobject is deactivated and later reactivated for example in a cutscene.